.github/workflows/ci.yml

```diff
@@ -0,0 +1,225 @@
+name: CI
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+  schedule:
+    - cron: '0 1 * * *'
+
+env:
+  RUSTFLAGS: -Dwarnings
+  RUST_BACKTRACE: 1
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  test:
+    name: cargo +${{ matrix.rust }} test (${{ matrix.os }})
+    strategy:
+      matrix:
+        rust:
+          - nightly
+        os:
+         - ubuntu-latest
+         - macos-latest
+         - windows-latest
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Rust
+        run: rustup update ${{ matrix.rust }} && rustup default ${{ matrix.rust }}
```

Comment:
Seems `rustup update ..` is does not actually need, as the recent rustup's `rustup default` automatically installs the toolchain if the toolchain is unavailable.

Reply:
Oh, no, this means if the default toolchain is stable and that is outdated, then the toolchain does not update. I'll revert...